### ISSUE TITLE
Validation before writing paper

### ISSUE THREAD

**Issue by mesarcik** [mesarcik/ROAD#18]
# Experiments for validation of models before committing to writing.
- [x] Determine which samples from each class we are misclassifying (if it seems unreasonable, then reassess model)
- [x] Obtain an estimate of how long inference takes per spectrogram
- [x] Evaluate OOD results in a more appropriate way 
    -  I think doing it on a per class level is a little weird
    - I think we should randomly drop a class train the classifier and see its overall performance at detecting anomalies 
    - Then we can do the same experiments with multiple classes, so see the overall performance wrt to number of classes dropped.
- [x] Dataset finalisation
    - Electric fence? 
    - Third order high noise fused with first order?  

# Results needed for paper:
The purpose of this paper is to show how we can solve the two problems, these being classification of known failures as well as anomaly detection for unseen anomalies.
 
1) Comparison of different anomaly detection models on our dataset vs our method.
    - Note this might be a little weird as vision-based AD models only evaluate in a single class setting.
    - I think the single class setting doesn't really make sense, so what we should compare against is fine-tuned ability to do anomaly detection. In the sense that we use all classes and we get it to predict if its anomalous or not. We could also technically do it without subsampling on all available data. 
 2) Comparison of classifier with and without SSL corrections that proves that our approach is best 
    - this will probably in the form of the bar graph done in the presentation where we show that we can get a 3% overall increase
3) Comparing the combined anomaly detection performance of the supervised backbone against the ssl augmentation when some classes are OOD
    - I envisage this to be a line chart with combined F-2 anomaly detection performance on the y-axis against number of OOD classes on the x-axis.  
4) How many epochs of pre-training/fine tuning repsectively to obtain best performance? 
5) Model ablations:
    - If we remove parts of the SSL loss we prescribe how does it effect the overall training objective in terms of anomaly detection.  
    - Combination function of classification and detection
